extensions/poetry/parse.ts

```diff
@@ -0,0 +1,36 @@
+import { parse } from "https://deno.land/std@0.148.0/flags/mod.ts"
+
+type PackageSpec = {
+  name: string,
+  version: string,
+}
+
+// Parse a package name and version from a line of a poetry dry run output.
+// Returns `null` if no match was found.
+function parseDryRunLine(line: string): PackageSpec | null {
+  const installingRegexp = /: selecting\s+([^\s]+)\s+\(([^\)]+)\)/
```

Comment:
The following command seems to print `selecting` for something that doesn't actually get installed. So I'm not sure your approach works:

```
$ poetry add --dry-run -vvv jinja
Creating virtualenv test-CHLv_kh8-py3.10 in /home/chris/.cache/pypoetry/virtualenvs
Using virtualenv: /home/chris/.cache/pypoetry/virtualenvs/test-CHLv_kh8-py3.10
PyPI: No release information found for jinja-0.5beta, skipping
PyPI: No release information found for jinja-0.6, skipping
PyPI: No release information found for jinja-0.7, skipping
PyPI: 5 packages found for jinja *
Using version ^1.2 for Jinja

Updating dependencies
Resolving dependencies...
   1: fact: test is 0.1.0
   1: derived: test
   1: fact: test depends on Jinja (^1.2)
   1: selecting test (0.1.0)
   1: derived: Jinja (>=1.2,<2.0)
PyPI: No release information found for jinja-0.5beta, skipping
PyPI: No release information found for jinja-0.6, skipping
PyPI: No release information found for jinja-0.7, skipping
PyPI: 1 packages found for jinja >=1.2,<2.0
   1: selecting jinja (1.2)
   1: Version solving took 0.008 seconds.
   1: Tried 1 solutions.

Writing lock file

Finding the necessary packages for the current system

Package operations: 1 install, 0 updates, 0 removals

  • Installing jinja (1.2)
  • Installing jinja (1.2)
```

Reply:
I tried to reproduce this with a project named `a` and I discovered that the first selected package is indeed `a`, so chances are your test package was named `test`:

```
Updating dependencies
Resolving dependencies...
   1: fact: a is 0.1.0
   1: derived: a
   1: fact: a depends on Jinja (^1.2)
   1: selecting a (0.1.0)
   1: derived: Jinja (>=1.2,<2.0)
PyPI: No release information found for jinja-0.5beta, skipping
PyPI: No release information found for jinja-0.6, skipping
PyPI: No release information found for jinja-0.7, skipping
PyPI: 1 packages found for jinja >=1.2,<2.0
   1: selecting jinja (1.2)
   1: Version solving took 0.004 seconds.
   1: Tried 1 solutions.
```

I repeat what I mentioned via DMs here. Parsing `Installing` lines is unreliable, because none is output with the `--lock` flag, but the lockfile is updated nonetheless:

```
$ poetry add --dry-run --lock -vvv jinja                                                                                     andrea phylum-ubuntu fsh 2022-08-02 13:35:25
Using virtualenv: /home/andrea/.cache/pypoetry/virtualenvs/a-2bdB6j1N-py3.10
PyPI: No release information found for jinja-0.5beta, skipping
PyPI: No release information found for jinja-0.6, skipping
PyPI: No release information found for jinja-0.7, skipping
PyPI: 5 packages found for jinja *
Using version ^1.2 for Jinja

Updating dependencies
Resolving dependencies...
   1: fact: a is 0.1.0
   1: derived: a
   1: fact: a depends on Jinja (^1.2)
   1: selecting a (0.1.0)
   1: derived: Jinja (>=1.2,<2.0)
PyPI: No release information found for jinja-0.5beta, skipping
PyPI: No release information found for jinja-0.6, skipping
PyPI: No release information found for jinja-0.7, skipping
PyPI: 1 packages found for jinja >=1.2,<2.0
   1: selecting jinja (1.2)
   1: Version solving took 0.003 seconds.
   1: Tried 1 solutions.
$
```

Reply:
This file was removed due to a reversion to submitting the whole package lockfile.